### ISSUE TITLE
Missing Filings -add column to view

### ISSUE THREAD

**Issue by jwchumley** [18F/openFEC#2723]
I think it is the API. The reports endpoint shows three rows for this committee with committee type S but filings endpoint shows none. So on the website you can find the reports under house and senate committee filings but not under all filings unless you don't specify the committee type and since this committee has been an H and is now an S you need to be able to specify.

Shows 3 reports:
https://www.fec.gov/data/reports/house-senate/?is_amended=false&data_type=processed&committee_id=C00501197&type=S

Shows no reports:
https://www.fec.gov/data/filings/?data_type=processed&committee_id=C00501197&office=S

/house-senate/ and /filings/ use different tables
- [ ] results should be the same - look at /filings/

**Comment by AmyKort** [comment]
@jwchumley Can you help me understand what this is?

**Comment by pkfec** [comment]
The two end points are calling different MV's. Should revisit the sql queries and identify where the discrepancy is coming from.

**/reports/house-senate/**

- ofec_reports_house_senate_mv


**/filings/**

- ofec_filings_mv

**Comment by LindsayYoung** [comment]
The problem is that they are missing from the filings view

**Comment by fecjjeng** [comment]
Is data_type='processed' translated into means_filed = 'paper' in both mv?

I don't see a committee_type column in ofec_reports_house_senate_mv.  Which column in ofec_reports_house_senate_mv correlated to the type='S'?  

Which column in ofec_filings_mv correlated to the office='S'?
If that is the "office" column responsible for the office='S' in ofec_filings_mv, then we will never get anything back if filtered by it as far as I can tell.  
Since none of the rows with a committee_id started with 'C' (that means a committee) has the office column filled.   

The following query returns 0  
select count(*) 
from ofec_filings_mv
where committee_id like 'C%'
and office is not null;

Have this filter ever successful in returning any rows if both the committee_id and the office filter had been selected?

**Comment by fecjjeng** [comment]
I had reviewed the ofec_filings_mv.  We can add a column cmte_tp and update the MV definition to get data from disclosure.cmte_valid_fec_yr.  The API can then filter on cmte_tp instead of office (which is empty)
However, the problem arise since the MV ofec_filings_mv has 15 other MVs depending on it (directly or indirectly, in different order), changes made to ofec_filings_mv will impact all the downstream MVs. 
Which make it very difficult to make the change in the production database to follow the 'zero-down-time' path in flyway migration script. 

--
-- MATERIALIZED VIEWs depending on public.ofec_filings_mv:
--
public.ofec_totals_candidate_committees_mv

public.ofec_totals_combined_mv
	public.ofec_candidate_fulltext_mv

	public.ofec_committee_fulltext_mv

	public.ofec_sched_a_aggregate_size_merged_mv

	public.ofec_totals_house_senate_mv
		public.ofec_candidate_totals_mv
			public.ofec_candidate_flag**
		public.ofec_entity_chart_mv*
	public.ofec_totals_ie_only_mv

	public.ofec_totals_pacs_parties_mv
		public.ofec_totals_pacs_mv
			public.ofec_entity_chart_mv*
		public.ofec_totals_parties_mv	
			public.ofec_entity_chart_mv*

	public.ofec_totals_presidential_mv
		public.ofec_candidate_totals_mv
			public.ofec_candidate_flag**

cc: @ccostino @LindsayYoung @jwchumley @PaulClark2 @vrajmohan

**Comment by LindsayYoung** [comment]
That is a good point @fecjjeng! 

Applying this change as a migration is trickier than what we had before. I would love to do a video chat so the team could come up with some strategies to minimize the amount of work and downtime it takes to make these kinds of changes.

**Comment by LindsayYoung** [comment]
We are going to do a more comprehensive fix but we will edit the view Priya needs directly to unblock her.

**Comment by pkfec** [comment]
During our local testing we discovered that filings should be filtered by committee type and not by office. And candidates are filtered by office. 

**What we fixed**: the filings data migration file *ofec_filings_all_mv* is now updated to return the filings based off committee_type. And the API filings model is updated with the committee_type field.

**This fixes the back end**. But there are some additional changes on the front-end. 

**What we need to do**:
We should have  committee type filter along the office filter for filings. I am going to open a new issue to address this.

**Comment by pkfec** [comment]
The new filings sql migration file , **ofec_filings_all_mv** is updated to return a new column **office_cmte_tp** for office or committee types H, S and P.

**office** column is model/filings.py is now mapped **office_cmte_tp** column to return the results.

Tested this fix locally and getting the expected results.

**Comment by LindsayYoung** [comment]
We made the "office" filter look for H, S or P for candidate filings and filings with committees that have the H, S or P committee_types.

So to be clear, this doesn't have joint committees or authorized committees. If that is something we want and we need to add those, that will be additional work. 

As an added bonus we can now filter by committee types on the API.

**Comment by fecjjeng** [comment]
@jwchumley would you please reopen issue #2723? I need to add in one more change to add the new MV to be included in the nightly refresh process.

**Comment by AmyKort** [comment]
merged, so closing.
